File: solace/cmd/play/Equipment.java
package solace.cmd.play;

import solace.net.*;
import solace.util.*;
import solace.game.*;

/**
 * Equipment command, displays a character's current equipment.
 * @author Ryan Sandor Richards
 */
public class Equipment extends PlayCommand {
  public Equipment(solace.game.Character ch) {
    super("equipment", ch);
  }

  public boolean run(Connection c, String []params) {
    StringBuffer b = new StringBuffer();
    b.append(Strings.banner("Equipment"));

    for (String slot : solace.game.Character.EQ_SLOTS) {
      Item item = character.getEquipment(slot);
      String name = (item == null) ?
        "---" : item.get("description.inventory");
      String line = String.format("| [ {y%-8s{x ]: %s", slot, name);
      b.append(line);
      b.append(Strings.spaces(80 - 1 - Color.strip(line).length()) + "|\n\r");
    }

    b.append(Strings.RULE);

    character.sendln(b.toString());

    return true;
  }
}

File: script/commands/inventory/equipment.js
'use strict';

var EQ_SLOTS = Packages.solace.game.Character.EQ_SLOTS;

/**
 * Equipment command, displays a character's current equipment.
 * @author Ryan Sandor Richards
 */
Commands.add('equipment', function (player, params) {
  var buffer = new StringBuffer();
  buffer.append(Strings.banner('Equipment'));

  EQ_SLOTS.forEach(function (slot) {
    var item = player.getCharacter().getEquipment(slot);
    var name = !item ? '---' : item.get('description.inventory');
    var line = format('| [ {y}%-8s{x} ]: %s', slot, name);
    buffer.append(line);
    buffer.append(Strings.spaces(
      80 - 1 - Color.strip(line).length()) + '|\n\r');
  });
  buffer.append(Strings.RULE);

  player.sendln(buffer.toString());
});
